src/components/PopupWithForm.js

```diff
@@ -29,8 +35,10 @@ class PopupWithForm extends Popup {
     }
   }
 
-  open() {
-    super.open();
+  close() {
+    super.close();
+    this._form.reset();
+    this.updateSubmitButtonText(this._submitButtonText);
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.